site-configs.ts

```diff
@@ -3,36 +3,44 @@ import { SiteConfig } from "./site-configs.d";
 
 // Types for files in site-configs/
 type Environment = "local" | "dev" | "test" | "staging" | "prod";
-export type Config = Omit<SiteConfig, "domains" | "public"> & SiteConfig["public"] & {
+export type Config = Omit<SiteConfig, "domains" | "public"> & Pick<SiteConfig["public"], "domain" | "languages"> & {
     domains: {
         preliminary?: string;
     } & {
         [key in Environment]?: string;
     };
 };
 
+const isLocalhost = (domain:string) => {
+    return domain.match(/\b(?:[a-zA-Z0-9-]+\.)?localhost:\d{4}\b/);
```

Comment:
I'm not sure what \b means here, I'd use ^ and $

```suggestion
    return domain.match(/^([a-zA-Z0-9-]+\.)?localhost:\d{4}$/);
```

Reply:
016068ba5900cba61c26aad6a71d14c23c174c35